src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-namespace-list/rbd-namespace-list.component.ts

```diff
@@ -72,18 +75,26 @@ export class RbdNamespaceListComponent implements OnInit {
   }
 
   refresh() {
-    this.rbdService.listNamespaces().subscribe((resp: any) => {
-      const namespaces = [];
-      resp.forEach((pool: any) => {
-        pool.namespaces.forEach((namespace) => {
-          namespaces.push({
-            id: `${pool.pool_name}/${namespace}`,
-            namespace: namespace,
-            pool: pool.pool_name
+    this.poolService.list(['pool_name']).then((pools: any) => {
```

Comment:
Just a minor comment: could we filter by pools with app rbd enabled? That should quickly avoid listing through non-RBD pools.

Reply:
Good catch. Fixed.